src/databricks/labs/remorph/transpiler/lsp/lsp_engine.py

```diff
@@ -229,6 +229,18 @@ def wrapper(params):
 
         return wrapper
 
+    async def start_io(self, cmd: str, *args, **kwargs):
+        await super().start_io(cmd, *args, **kwargs)
+        # forward stderr
+        task = asyncio.create_task(self.pipe_stderr())
+        self._async_tasks.append(task)
+
+    async def pipe_stderr(self):
+        while not self._stop_event.is_set():
```

Comment:
This loop should also exit if you read a line that doesn't end in `\n`: that indicates EOF has been received. In general it would be okay for an LSP server to close stderr if it didn't plan to use it, and as things stand that would lead to a busy loop of infinite logging.

Reply:
I believe that the server closing stderr would raise an exception on the client side, but it doesn't hurt. Done